From 0fc23507c141c2f81db89c568eb9620dc5b6e2df Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Wed, 3 Apr 2019 02:15:29 +0000
Subject: [PATCH 1/3] spp_nfv: remove declaration in for loop
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

For some environment, declaration in for loop cause an error.

  error: ‘for’ loop initial declarations are only allowed in C99 mode

This update is move this declaration from.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 src/nfv/main.c       | 2 +-
 src/nfv/nfv_status.c | 4 ++--
 src/nfv/nfv_utils.h  | 3 ++-
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/nfv/main.c b/src/nfv/main.c
index 4ecc553..fbcef0c 100644
--- a/src/nfv/main.c
+++ b/src/nfv/main.c
@@ -248,7 +248,7 @@ main(int argc, char *argv[])
 		lcore_id_used[lcore_id] = 1;
 	}
 	sprintf(log_msg, "Used lcores: ");
-	for (int i = 0; i < RTE_MAX_LCORE; i++) {
+	for (i = 0; i < RTE_MAX_LCORE; i++) {
 		if (lcore_id_used[i] == 1)
 			sprintf(log_msg + strlen(log_msg), "%d ", i);
 	}
diff --git a/src/nfv/nfv_status.c b/src/nfv/nfv_status.c
index a3ad597..d29e083 100644
--- a/src/nfv/nfv_status.c
+++ b/src/nfv/nfv_status.c
@@ -54,9 +54,9 @@ int
 append_lcore_info_json(char *str,
 		uint8_t lcore_id_used[RTE_MAX_LCORE])
 {
-
+	int i;
 	sprintf(str + strlen(str), "\"lcores\":[");
-	for (int i = 0; i < RTE_MAX_LCORE; i++) {
+	for (i = 0; i < RTE_MAX_LCORE; i++) {
 		if (lcore_id_used[i] == 1)
 			sprintf(str + strlen(str), "%d,", i);
 	}
diff --git a/src/nfv/nfv_utils.h b/src/nfv/nfv_utils.h
index aca5f13..9d4f9dd 100644
--- a/src/nfv/nfv_utils.h
+++ b/src/nfv/nfv_utils.h
@@ -107,7 +107,8 @@ forward_array_reset(void)
 /* Return a type of port as a enum member of porttype_map structure. */
 static enum port_type get_port_type(char *portname)
 {
-	for (int i = 0; portmap[i].port_name != NULL; i++) {
+	int i;
+	for (i = 0; portmap[i].port_name != NULL; i++) {
 		const char *port_name = portmap[i].port_name;
 		if (strncmp(portname, port_name, strlen(port_name)) == 0)
 			return portmap[i].port_type;
-- 
2.7.4

